File: android/src/main/java/com/azifacemobile/theme/SessionAbortAnimation.java
package com.azifacemobile.theme;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONObject;

public class SessionAbortAnimation extends ViewStyle {
  private final static String KEY = "sessionAbortAnimation";
  private final Image image;

  public SessionAbortAnimation(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    JSONObject currentTarget = new Theme().getTarget(KEY);

    this.image = new Image(context, currentTarget);
  }

  @Override
  public int getBackgroundColor() {
    return super.getBackgroundColor("#CC0044");
  }

  public int getBackgroundImage() {
    return this.image.getSource("image");
  }
}

File: ios/Theme/SessionAbortAnimation.swift
public class SessionAbortAnimation: ViewStyle {
  private static var KEY: String = "sessionAbortAnimation"
  private var image: Image

  init(target: NSDictionary?) {
    let currentTarget = Style().getTarget(target, key: SessionAbortAnimation.KEY)
    
    self.image = Image(target: currentTarget)
    
    super.init(target: target, key: SessionAbortAnimation.KEY)
  }

  override public func getBackgroundColor() -> UIColor {
    return self.getBackgroundColor(defaultColor: "#CC0044")
  }
  
  public func getBackgroundImage() -> UIImage? {
    return self.image.getSource("image")
  }
}
